File: src/2315.count-asterisks.2401/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var countAsterisks = function (s) {
    let result = 0;
    let inner = false;
    for (const c of s) {
        if (c === '|') {
            inner = !inner;
        } else if (!inner && c === '*') {
            result++;
        }
    }
    return result;
};

File: src/2315.count-asterisks.2401/solution.java
class Solution {
    public int countAsterisks(String s) {
        char[] chars = s.toCharArray();
        boolean status = false;
        int result = 0;
        for (char c : chars) {
            if (c == '|') {
                status = !status;
            } else if (!status && c == '*') {
                result++;
            }

        }
        return result;
    }
}